src/h5web/vis-packs/core/utils.ts

```diff
@@ -42,11 +41,20 @@ const adaptedLogTicksThreshold = scaleLinear({
 });
 
 export function createAxisScale(
-  config: PickScaleConfig<ScaleType, number>
+  scaleParams: ScaleParams,
+  config: VisxScaleConfig | GammaScaleConfig
 ): AxisScale {
-  const { type } = config;
+  if (isScaleType(scaleParams) && scaleParams !== ScaleType.Gamma) {
+    return H5WEB_SCALES[scaleParams].createScale(config);
+  }
 
-  return H5WEB_SCALES[type].createScale(config);
+  const [scaleType, exponent] = scaleParams;
+  if (scaleType !== ScaleType.Gamma) {
+    throw new Error(
+      'scaleParams should be an array only for Gamma scale type !'
+    );
+  }
```

Comment:
Same, I don't get how this can happen based on the type of `VisScaleType`.

Reply:
https://github.com/silx-kit/h5web/pull/760#discussion_r686076029 and this were to help TypeScript which really struggles to infer the types.

It does not matter that much in the end as `createScale` accepts `any`.